desc/plotting.py

```diff
@@ -626,8 +635,62 @@ def plot_2d(
     plot_axes = _get_plot_axes(grid)
     if len(plot_axes) != 2:
         return ValueError(colored("Grid must be 2D", "red"))
+    component = kwargs.pop("component", None)
+    if name != "B*n":
+        data, label = _compute(
+            eq,
+            name,
+            grid,
+            component=component,
+        )
+    else:
+        field = kwargs.pop("field", None)
+        errorif(
+            field is None,
+            ValueError,
+            "If Bn is entered as the variable to plot, a magnetic field"
+            " must be provided.",
+        )
+        field_grid = kwargs.pop("field_grid", None)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            data, _ = field.compute_Bnormal(eq, eval_grid=grid, source_grid=field_grid)
+            vc_data = eq.compute(
+                [
+                    "K_vc",
+                    "B",
+                    "R",
+                    "phi",
+                    "Z",
+                    "e^rho",
+                    "n_rho",
+                    "|e_theta x e_zeta|",
+                ],
+                grid=grid,
+            )
+            k = min(grid.num_theta, grid.num_zeta * grid.NFP)
+            s = k - 1
+            k = min(grid.num_theta, grid.num_zeta * grid.NFP)
+            q = k // 2 + int(np.sqrt(k))
+            try:
+                interpolator = FFTInterpolator(grid, grid, s, q)
+            except AssertionError:
+                interpolator = DFTInterpolator(grid, grid, s, q)
+            if hasattr(eq.surface, "Phi_mn"):
+                vc_data["K_vc"] += eq.surface.compute("K", grid=grid)["K"]
+            Bplasma = virtual_casing_biot_savart(
+                vc_data,
+                vc_data,
+                interpolator,
+            )
+            # need extra factor of B/2 bc we're evaluating on plasma surface
```

Comment:
technically we don't need this since the equilibrium B always has zero normal component

Reply:
Ah is this only needed for the pressure balance?

Reply:
yeah